src/Sulu/Bundle/ContactBundle/Infrastructure/Sulu/Search/AccountReindexProvider.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\ContactBundle\Infrastructure\Sulu\Search;
+
+use CmsIg\Seal\Reindex\ReindexConfig;
+use CmsIg\Seal\Reindex\ReindexProviderInterface;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\EntityRepository;
+use Sulu\Bundle\ContactBundle\Entity\AccountInterface;
+
+class AccountReindexProvider implements ReindexProviderInterface
+{
+    /**
+     * @var EntityRepository<AccountInterface>
+     */
+    protected EntityRepository $accountRepository;
+
+    public function __construct(
+        EntityManagerInterface $entityManager,
+    ) {
+        $repository = $entityManager->getRepository(AccountInterface::class);
+
+        $this->accountRepository = $repository;
+    }
+
+    public function total(): ?int
+    {
+        return null;
+    }
+
+    public function provide(ReindexConfig $reindexConfig): \Generator
+    {
+        $accounts = $this->loadAccounts($reindexConfig->getIdentifiers());
+
+        foreach ($accounts as $account) {
+            yield [
+                'id' => AccountInterface::RESOURCE_KEY . '::' . $account['id'],
+                'resourceKey' => AccountInterface::RESOURCE_KEY,
+                'resourceId' => (string) $account['id'],
+                'changedAt' => $account['changed'] ? $account['changed']->format('c') : null,
+                'createdAt' => $account['created'] ? $account['created']->format('c') : null,
+                'title' => $account['name'],
+            ];
+        }
+    }
+
+    /**
+     * @param string[] $identifiers
+     */
+    private function loadAccounts(array $identifiers = []): \Generator
+    {
+        $qb = $this->accountRepository->createQueryBuilder('account')
+            ->select('account.id')
+            ->addSelect('account.name')
+            ->addSelect('account.created')
+            ->addSelect('account.changed');
+
+        if (0 < \count($identifiers)) {
+            $qb->where('account.id IN (:ids)')
+                ->setParameter('ids', \array_map(fn ($identifier) => (int) \str_replace(AccountInterface::RESOURCE_KEY . '::', '', $identifier), $identifiers));
+        }
+
+        return $qb->getQuery()->toIterable();
```

Comment:
You can workaround the phpstan error this way:

```suggestion
        foreach ($qb->getQuery()->toIterable() as $row) {
            yield $row;
        }
```